test/e2e/operator_test.go

```diff
@@ -0,0 +1,279 @@
+package e2e
+
+import (
+	"fmt"
+	"math/rand"
+	"strconv"
+	"testing"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	framework "github.com/openshift/cluster-csi-snapshot-controller-operator/test/framework"
+	"github.com/openshift/cluster-csi-snapshot-controller-operator/test/helpers"
+	kapierrs "k8s.io/apimachinery/pkg/api/errors"
+
+	"k8s.io/kubernetes/test/e2e/storage/drivers"
+
+	volumesnapshotsv1beta1 "github.com/kubernetes-csi/external-snapshotter/v2/pkg/apis/volumesnapshot/v1beta1"
+)
+
+const (
+	snapshotGroup      = "snapshot.storage.k8s.io"
+	snapshotAPIVersion = "snapshot.storage.k8s.io/v1beta1"
+
+	operatorNamespace = "openshift-csi-snapshot-controller-operator"
+	operatorName      = "csi-snapshot-controller-operator"
+)
+
+var (
+	retryInterval        = time.Second * 5
+	timeout              = time.Second * 120
+	cleanupRetryInterval = time.Second * 1
+	cleanupTimeout       = time.Second * 5
+	poll                 = 2 * time.Second
+	size                 = 1000000
+)
+
+func TestCSISnapshotControllerOperator(t *testing.T) {
+	ctx := framework.NewClientSet("")
+	driver := drivers.InitHostPathCSIDriver()
+
+	// Create a namespace for the subsequent objects
+	namespace, err := createNamespace(t, ctx)
+	if err != nil {
+		t.Fatalf("Unable to create a corresponding namespace: %v", err)
+	}
+	defer func() {
+		err = ctx.CoreV1Interface.Namespaces().Delete(namespace, &metav1.DeleteOptions{})
+		if err != nil {
+			t.Fatalf("Error attempting to delete the namespace: %v", err)
+		}
+	}()
+
+	// Ensure that the Operator is ready
+	err = waitForOperatorToBeReady(t, ctx)
+	if err != nil {
+		t.Fatalf("ClusterOperator never became ready: %v", err)
+	}
+
+	// Ensure the VolumeSnapshot CRD is installed
+	// I wonder if this is even useful, and if it should be removed, since we check that the Operator is available?
+	_, err = ctx.ApiextensionsV1beta1Interface.CustomResourceDefinitions().Get("volumesnapshots.snapshot.storage.k8s.io", metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("Error attempting to retrieve snapshot CRD: %v", err)
+	}
+
+	// Create the PV
+	pv := helpers.CreateFakePV(t, "pv"+strconv.Itoa(rand.Intn(1000)), driver.GetDriverInfo().Name, "pv", size)
```

Comment:
nit: The pv isn't really fake, so you might want to rename this function.

Also, I think `t` isn't used anywhere. The pv name could be moved into the func as well

Reply:
`t` was originally used so I could log when this function did more. I've removed the reference, and moved all of the name creation into the functions themselves.